src/ports/postgres/modules/utilities/pivot.py_in

```diff
@@ -234,30 +235,46 @@ def pivot(schema_madlib, source_table, out_table, index, pivot_cols,
                     dict_counter += 1
                 # Collecting the whole sql query
                 # Please refer to the earlier comment for a sample output
-                pivot_str = ("{fill_str_begin}"
-                             "{agg} (CASE WHEN {pivot_begin} THEN {pval} END) "
-                             "{fill_str_end} AS {pivot_end}".
-                             format(fill_str_begin=fill_str_begin,
+
+                # Create the masked set
+                pivot_str_from = ("(CASE WHEN {pivot_begin} THEN {pval} END) "
+                            "AS {pivot_end}".
+                            format(fill_str_begin=fill_str_begin,
+                                    fill_str_end=fill_str_end,
+                                    pval=pval, agg=agg,
+                                    pivot_begin=' AND '.join(pivot_str_begin),
+                                    pivot_end=''.join(pivot_str_end)))
+                pivot_str_from_list.append("," + pivot_str_from)
+                # Apply the aggregate and filtering
+                pivot_str_sel = ("{fill_str_begin}"
+                            "{agg} ({pivot_end}) FILTER ("
+                            "WHERE {pivot_end} IS NOT NULL) "
+                            "{fill_str_end} AS {pivot_end}".
+                            format(fill_str_begin=fill_str_begin,
                                     fill_str_end=fill_str_end,
                                     pval=pval, agg=agg,
                                     pivot_begin=' AND '.join(pivot_str_begin),
                                     pivot_end=''.join(pivot_str_end)))
-                pivot_out_list.append("," + pivot_str)
+                pivot_str_sel_list.append("," + pivot_str_sel)
```

Comment:
Would be nicer to only add `pivot_str_sel` in the list and let `join` take care of adding the commas.